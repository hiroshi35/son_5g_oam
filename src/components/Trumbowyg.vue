<template>
  <textarea>{{value}}</textarea>
</template>

<script>
  import 'trumbowyg'
  import 'trumbowyg/plugins/base64/trumbowyg.base64.js'
  import 'trumbowyg/plugins/colors/trumbowyg.colors.js'
  import 'trumbowyg/plugins/table/trumbowyg.table.js'
  import 'trumbowyg/plugins/upload/trumbowyg.upload.js'

  export default {
    props: {
      autogrow: {
        type: Boolean,
        default: false
      },
      options: {
        type: Object
      },
      uploadPath: {
        type: String
      },
      value: {
        type: String
      }
    },
    computed: {
      optionsObj () {
        if (this.options) {
          return this.options;
        }
        return {
          resetCss: true,
          plugins: {
            upload: {
              serverPath: this.uploadPath
            }
          },
          autogrow: this.autogrow,
          btnsDef: {
            image: {
              dropdown: ['insertImage', 'base64', 'upload'],
              ico: 'insertImage'
            }
          },
          btns: [
            ['viewHTML'],
            ['formatting'],
            ['strong', 'em'],
            'btnGrp-semantic',
            ['superscript', 'subscript'],
            ['link'],
            ['image'],
            'btnGrp-justify',
            'btnGrp-lists',
            ['horizontalRule'],
            ['removeformat'],
            ['foreColor', 'backColor'],
            ['table'],
            ['fullscreen']
          ]
        }
      }
    },
    mounted () {
      let vm = this
      $.trumbowyg.svgPath = require('trumbowyg/dist/ui/icons.svg');
      $(this.$el).trumbowyg(this.optionsObj)
      $(this.$el).trumbowyg().on('tbwchange', (e) => {
        vm.$emit('input', e.target.value)
      })
      $(this.$el).trumbowyg().on('tbwinit', (e) => {
        vm.$emit('mounted')
      })
    }
  }
</script>
<style lang="scss">
  @import "~trumbowyg/dist/ui/trumbowyg";
  @import "~trumbowyg/dist/plugins/colors/ui/trumbowyg.colors";

  #trumbowyg-icons {
    display: none;
  }
</style>


