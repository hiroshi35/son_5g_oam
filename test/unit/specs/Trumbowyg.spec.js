import { shallow } from 'vue-test-utils'
import Trumbowyg from '@/components/Trumbowyg.vue'
import {trumbowyg, trumbowygOn} from 'jquery'

describe('Trumbowyg.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Trumbowyg)
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  test('$().trumbowyg() should be called with correct options', () => {
    expect(trumbowyg).toBeCalledWith(wrapper.vm.optionsObj)
  })
  test('$().trumbowyg().on should be called with correct arguments', () => {
    expect(trumbowygOn.mock.calls[0][0]).toBe('tbwchange')
    expect(trumbowygOn.mock.calls[1][0]).toBe('tbwinit')
  })
})
