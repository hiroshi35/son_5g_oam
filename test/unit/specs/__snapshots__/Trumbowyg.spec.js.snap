// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trumbowyg.vue has the expected html structure 1`] = `
<textarea>
  
</textarea>
`;
